src/cpp/fastdds/xtypes/serializers/json/dynamic_data_json.cpp

```diff
@@ -744,9 +746,20 @@ ReturnCode_t json_serialize_basic_member(
             if (RETCODE_OK == ret)
             {
                 // Insert UTF-8 converted value
+#if defined(MINGW_COMPILER)
+                std::wstring aux_wstring_value({value});
+                std::string utf8_value;
+                int size_needed = std::wcstombs(nullptr, aux_wstring_value.data(), 0);
+                if (size_needed > 0) {
+                    utf8_value.resize(size_needed);
+                    std::wcstombs(&utf8_value[0], aux_wstring_value.data(), size_needed);
+                }
+#else
                 std::wstring aux_wstring_value({value});
                 std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
                 std::string utf8_value = converter.to_bytes(aux_wstring_value);
+
+#endif  // defined(MINGW_COMPILER)              
```

Comment:
Same empty space here. I know can be seen irrelevant change, but it's needed to pass uncrustify CI
```suggestion
#endif  // defined(MINGW_COMPILER)
```